### PR TITLE
Archived rules checkbox and text

### PR DESCRIPTION
**Pull request by jeoffreyfischer**
**Pain**
As per my conversation with @adamcogan, the archived rules are only accessible on a separate page. 
It would be good to see all the rules in the same page (archived and unarchived).

**Solution**
A checkbox has been added at the top of the page to display archived rules.
The archived rules have a "archived" tag to be differentiated from the other rules.

Relates to #1196 

**Screenshots**
rules-PR-003a
**Figure**: The new "Include archived" checkbox.

rules-PR-003b
**Figure**: When clicked, the archived rules are listed with a "Archived" badge.

### CODE REVIEW

File: src/templates/category.js
Comment (JackDevAU) [inline src/templates/category.js:158]: Cool feature! 
I'm not sure if we should make archived rules more searchable? We currently have https://www.ssw.com.au/rules/archived/

But would like to hear @tiagov8 thoughts!

File: src/templates/category.js
Comment (JackDevAU) [inline src/templates/category.js:160]: lets make it red!

**Review by JackDevAU — CHANGES_REQUESTED** [review]

File: src/templates/category.js
Comment (jaydenalchin) [reply src/templates/category.js:158]: I agree with @JackDevAU

Comment (jeoffreyfischer) [reply src/templates/category.js:158]: Just for context, this task was assigned to me by Adam

File: src/templates/category.js
Comment (jeoffreyfischer) [reply src/templates/category.js:160]: Done - the checkbox and badge are in ssw-red

File: src/templates/category.js
Comment (tiagov8) [reply src/templates/category.js:158]: I agree with @JackDevAU and @jaydenalchin 

This feature gives unnecessary visibility to outdated content.

I know @adamcogan asked for it... but what is the goal with this?

Comment (jeoffreyfischer) [reply src/templates/category.js:158]: As per my conversation with @adamcogan, this feature is turned off by default, so it is not visible at page load. The goal is allow user to view archived rules if they want to.

Comment (tiagov8) [reply src/templates/category.js:158]: > we are migrating rules and deciding whether they should be archived or not.

I think this decision should be made **before** migrating. 

We already have hundreds of archived rules... 
1. How are we going to differentiate the "real archived" (decision already made) from the "to be archived or not" (recently migrated, decision to be made)?

> Maybe once all the rules are migrated we can remove the feature.

I'highly skeptical this will ever happen :/

---

In summary, I can live with the extra visibility for archived rules because @adamcogan wants it, but the reasons given and the process of **migrate first, decide later** are not ideal from my perspective. AKA Check rules before migrating...

Comment (tiagov8) [reply src/templates/category.js:158]: Archived rules generally goes into 3 scenarios:
A. Content that got **outdated** and cannot be updated (e.g. A software we used but got deprecated)
B. Content that is **duplicated**. 
C. Content that is **irrelevant**. (either was badly written, was set as TODO and never done)

**A** should exist (be visible to the public as long as they indicate they are outdated. They have value as history).
**B** are OK to exist, but should be avoided as much as possible (they cause unnecessary maintenance costs).
**C** should not exist.

I think we should try to prevent **some B** and **all the C** to be migrated.

Comment (jeoffreyfischer) [reply src/templates/category.js:158]: Hi @tiagov8 

     >1. How are we going to differentiate the "real archived" (decision already made) from the "to be archived or not" (recently migrated, decision to be made)?
 The decision to archive or not a rule is made beforehand. I only migrate rules after they are approved by @adamcogan. Sorry if this wasn't clear.

Comment (jaydenalchin) [reply src/templates/category.js:158]: Thanks for following up on this issue @jeoffreyfischer 

@adamcogan @tiagov8 for the record I very strongly feel this introduces serious content issues and awful UX as discussed above.

We have an infamous [SugarLearning item](https://my.sugarlearning.com/SSW/items/13061/sharp-eye-find-the-error) about how outdated content is embarrassing. We shouldn't be making it this visible. 

> The goal is allow user to view archived rules if they want to.

This goal is completely understandable but it has already been achieved - see the [Archived Rules page](https://www.ssw.com.au/rules/archived/). 

If we want to make it easier for curious users to find _specific_ archived rules, a better solution would be making the existing Archived Rules page more searchable and filterable instead.

Comment (tiagov8) [reply src/templates/category.js:158]: +1 JD

This feature is against good practices of 'less is more'

Comment (jeoffreyfischer) [reply src/templates/category.js:158]: @tiagov8 @jaydenalchin 
Yes you are right, the only advantage I saw in having this feature is to be able to check whether we have duplicate rules during the migration process.
But I realise that I can check that by opening the archived page and do a quick search. It will take a little more time but that's negligible compared to the drawback of giving unnecessary visibility to archived rules, as stated by Tiago.

Comment (tiagov8) [reply src/templates/category.js:158]: @jeoffreyfischer 

What's the final decision? Can you confirm with PO we ca revert this change and cancel the request?

Comment (jeoffreyfischer) [reply src/templates/category.js:158]: As per my conversation with @adamcogan, he explained to me that this feature is actually important because we currently do not have a direct access from a rules category page to the related archived rules.
As some concern was raised in this conversation by @tiagov8 , @jaydenalchin  and @JackDevAU , once the feature is live I will run analytics on it to see whether it's being used or not. 
If we realise that it is unpopular, we will discuss its use again.
I have prepared a "To myself" email that I will send to all of us once the feature is live, subject "Rules category - run analytics on buttons and checkbox"

Comment (tiagov8) [reply src/templates/category.js:158]: Hi @jeoffreyfischer / @adamcogan 

See this related Isuue comment: https://github.com/SSWConsulting/SSW.Rules/issues/934#issuecomment-1435140403 (the full issue is quite long)

Basically if we make the Archived page to show rules by category (as the regular rules), we solve the problem without having to mix archived/live rules up...

IMO it is a way better solution!

Let me know if in doubt. 

---

> once the feature is live I will run analytics on it to see whether it's being used or not.
> If we realise that it is unpopular, we will discuss its use again.

I am pessimistic about the possibilites with this... no matter the results, this feature will be there for good 🙁 I think the best way to avoid a bad feature is to not implement it.

Comment (jaydenalchin) [reply src/templates/category.js:158]: For the record if this has to be a feature I'd prefer it be a link to the Archived Rules page at the bottom of a category e.g. "Can't find what you're looking for? [See Archived Rules]()".

Comment (JackDevAU) [reply src/templates/category.js:158]: I'm with Jayden and Tiago on this one, once this feature is in it'll never be removed again (until the rewrite of Rules 😅) But as this is something the Product Owner wants - I can't block this feature being implemented :P

**Review by JackDevAU — APPROVED** [review]

File: src/templates/category.js
Comment (jeoffreyfischer) [reply src/templates/category.js:158]: As per my conversation with @adamcogan, I explained to him the alternative solutions proposed by @jaydenalchin and @tiagov8 and proposed him to set a meeting with all of us to discuss this. 
He decided to keep going with the initial plan i.e. the checkbox to display archived rules. The reason being that none of the 2 options provide a direct access from a rules category page to the respective archived rules category. 
Given this requirement, I also agree that the initial solution is the best.

**Review by jaydenalchin — APPROVED** [review]
Approving as the Product Owner is adamant - for the record I still think this is a terrible feature.

**Pull request merged**
